test-kit/components/date-picker-driver.ts

```diff
@@ -0,0 +1,84 @@
+import {DriverBase, selectDom, simulate, trigger} from 'test-drive-react';
+import {DatePicker} from '../../src';
+import {getDayNames} from '../../src/utils';
+
+const bodySelect = selectDom(document.body);
+const datePickerDropdown = 'DATE_PICKER_DROPDOWN';
+
+export class DatePickerTestDriver extends DriverBase {
+    public static ComponentClass = DatePicker;
+
+    public get input(): HTMLInputElement {
+        return this.select<HTMLInputElement>('DATE_PICKER_INPUT');
+    }
+
+    public get selectedDate(): string {
+        return this.select<HTMLInputElement>('DATE_PICKER_INPUT').value;
+    }
+
+    public changeDate(value: string): void {
+        trigger.change(this.input, value);
+        simulate.blur(this.input);
+    }
+
+    public click(elem?: Element): void {
```

Comment:
This should be separated into several methods (the names are just rough suggestions, but you get the idea):
 * clickOnDatePicker()
 * clickOnDay(x)
 * clickOnNextMonth()
 * clickOnPrevMonth()

Otherwise you exposing the dom structure outside the driver